documentation/prometheus-mixin/alerts.libsonnet

```diff
@@ -20,6 +20,20 @@
               description: 'Prometheus %(prometheusName)s has failed to reload its configuration.' % $._config,
             },
           },
+          {
+            alert: 'PrometheusSDRefreshFailure',
+            expr: |||
+              rate(prometheus_sd_refresh_failures_total{%(prometheusSelector)s}[5m]) > 0
```

Comment:
Let's use `increase` here. Then the alert value will tell you how many refresh failures there have been in the last 5m. With `rate`, you'll get a rather small number of "refresh failures per second".

Reply:
About the range: Some SD mechanisms have rather long default refresh intervals (e.g. 5m), which might lead to false negatives (this expression will go on and off, so that the `for` clause will prevent the alert from ever firing.

Maybe use a range of 10m here and 20m in the `for` clause?

Reply:
About the range: Some SD mechanisms have rather long default refresh intervals (e.g. 5m), which might lead to false negatives (this expression will go on and off, so that the `for` clause will prevent the alert from ever firing.

Maybe use a range of 10m here and 20m in the `for` clause?

Reply:
Thanks for your suggestion, I believe it's all done now.